File: php/src/Application/Catalogue/ListItemFamilies.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\Entity\ItemFamily;

final readonly class ListItemFamilies
{
    public function __construct(private ItemFamilyRepository $repository)
    {
    }

    /** @return list<ItemFamily> */
    public function execute(): array
    {
        return $this->repository->findAll();
    }
}

File: java/src/main/java/com/alxarafe/app/application/catalogue/ListItemFamilies.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.entity.ItemFamily;

import java.util.List;

public final class ListItemFamilies {
    private final ItemFamilyRepository repository;

    public ListItemFamilies(ItemFamilyRepository repository) {
        this.repository = repository;
    }

    public List<ItemFamily> execute() {
        return repository.findAll();
    }
}